speech/snippets/multichannel.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2016 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+// Command caption sends multichannel audio data to the Google Speech API
+// and prints its transcript.
+// go run multichannel.go ../testdata/commercial_stereo.wav
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+
+	speech "cloud.google.com/go/speech/apiv1p1beta1"
+	speechpb "google.golang.org/genproto/googleapis/cloud/speech/v1p1beta1"
+)
+
+const usage = `Usage: multichannel <audiofile>
+
+Audio file must be a 16-bit signed stereo (2-channel) with a sample rate of 44100.
+
+The path to the audio file may be a GCS URI (gs://...).
+`
+
+func main() {
+	flag.Parse()
+	if len(os.Args) < 2 {
+		fmt.Fprintln(os.Stderr, usage)
+		os.Exit(2)
+	}
+
+	var runFunc func(io.Writer, string) error
+
+	path := os.Args[1]
+	if strings.Contains(path, "://") {
```

Comment:
Could this make the function calls directly rather than using `runFunc`?

Reply:
created a direct call